src/client/partition.rs

```diff
@@ -97,6 +97,10 @@ pub enum OffsetAt {
 
     /// The latest existing record.
     Latest,
+
+    #[cfg(feature = "timestamp-query")]
+    /// Timestamp
+    Timestamp(i64),
```

Comment:
High-level APIs should have strong types. `i64` is not a timestamp, it's a number. So the user doesn't know what the semantics of this is (my guess is that it's something like "milliseconds from unix epoch", but you have to check with the Kafka docs to be sure). Since `rskafka` already depends on `chrono`, `DateTime<Utc>` would be a good candidate.